kotti-tests-kovenant/src/main/kotlin/com/github/adeynack/kotti/tests/kovenant/PromiseAssertions.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.adeynack.kotti.tests.kovenant
+
+import com.github.adeynack.kotti.tests.AssertionUtilities.fail
+import com.github.adeynack.kotti.tests.AssertionUtilities.isA
+import nl.komponents.kovenant.Promise
+import org.amshove.kluent.ExceptionResult
+import kotlin.reflect.KClass
+
+object PromiseAssertions {
+
+    infix fun <V, E : Exception> Promise<V, E>.shouldFailWith(expectedFailureType: KClass<out E>): ExceptionResult<E> {
```

Comment:
@mkulak : Just tried it and because of the type signature, we are forced to specify 3 types just to make it work:

```kotlin
val result = p.shouldFailWith<Int, Exception, ArithmeticException>()
```

Not super pretty 😞